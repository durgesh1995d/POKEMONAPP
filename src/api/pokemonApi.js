import { createApi, fetchBaseQuery } from '@reduxjs/toolkit/query/react';

export const pokemonApi = createApi({
  reducerPath: 'pokemonApi',
  baseQuery: fetchBaseQuery({ baseUrl: 'https://pokeapi.co/api/v2/' }),
  endpoints: (builder) => ({
    getPokemonList: builder.query({
      query: () => 'pokemon',
    }),
    getPokemonByName: builder.query({
      query: (name) => `pokemon/${name}`,
    }),
  }),
});
export const { useGetPokemonListQuery, useGetPokemonByNameQuery } = pokemonApi;
