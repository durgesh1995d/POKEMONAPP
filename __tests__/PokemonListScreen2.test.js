import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import PokemonList from '../src/component/PokemonList';
import { useGetPokemonListQuery } from '../src/api/pokemonApi';

// Mock the API hook
jest.mock('../src/api/pokemonApi', () => ({
  useGetPokemonListQuery: jest.fn(),
}));

const mockNavigate = jest.fn();

const mockData = {
  results: [{ name: 'Bulbasaur' }, { name: 'Charmander' }],
};

describe('PokemonList', () => {
  it('displays loading state correctly', () => {
    useGetPokemonListQuery.mockReturnValue({ isLoading: true });
    const { getByText,toJSON } = render(
      <PokemonList navigation={{ navigate: mockNavigate }} />
    );
    expect(getByText('Loading...')).toBeTruthy();
  expect(toJSON()).toMatchSnapshot()
  });

  it('displays error state correctly', () => {
    useGetPokemonListQuery.mockReturnValue({ error: true, isLoading: false });
    const { getByText,toJSON } = render(
      <PokemonList navigation={{ navigate: mockNavigate }} />
    );
    expect(getByText('Error loading data')).toBeTruthy();
    expect(toJSON()).toMatchSnapshot()
  });

  it('renders list of pokemons correctly', () => {
    useGetPokemonListQuery.mockReturnValue({
      data: mockData,
      isLoading: false,
    });
    const { getByText,toJSON } = render(
      <PokemonList navigation={{ navigate: mockNavigate }} />
    );
    expect(getByText('Bulbasaur')).toBeTruthy();
    expect(getByText('Charmander')).toBeTruthy();
  expect(toJSON()).toMatchSnapshot()
  });

  it('navigates on press', () => {
    useGetPokemonListQuery.mockReturnValue({
      data: mockData,
      isLoading: false,
    });
    const { getByText,toJSON } = render(
      <PokemonList navigation={{ navigate: mockNavigate }} />
    );
    fireEvent.press(getByText('Bulbasaur'));
    expect(mockNavigate).toHaveBeenCalledWith('PokemonDetail', {
      name: 'Bulbasaur',
    });
    expect(toJSON()).toMatchSnapshot()
  });
});
