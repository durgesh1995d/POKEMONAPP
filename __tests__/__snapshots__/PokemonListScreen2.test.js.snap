// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonList displays error state correctly 1`] = `
<Text>
  Error loading data
</Text>
`;

exports[`PokemonList displays loading state correctly 1`] = `
<Text>
  Loading...
</Text>
`;

exports[`PokemonList navigates on press 1`] = `
<RCTScrollView
  data={
    [
      {
        "name": "Bulbasaur",
      },
      {
        "name": "Charmander",
      },
    ]
  }
  getItem={[Function]}
  getItemCount={[Function]}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={0.0001}
  stickyHeaderIndices={[]}
  testID="PokemonList"
  viewabilityConfigCallbackPairs={[]}
>
  <View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "borderColor": "#666",
            "borderTopWidth": 1,
            "flexDirection": "row",
            "justifyContent": "space-around",
            "opacity": 1,
            "padding": 15,
          }
        }
      >
        <Image
          source={
            {
              "uri": undefined,
            }
          }
          style={
            {
              "resizeMode": "cover",
            }
          }
        />
        <Text
          style={
            {
              "color": "#000",
              "fontSize": 22,
            }
          }
        >
          Bulbasaur
        </Text>
      </View>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "borderColor": "#666",
            "borderTopWidth": 1,
            "flexDirection": "row",
            "justifyContent": "space-around",
            "opacity": 1,
            "padding": 15,
          }
        }
      >
        <Image
          source={
            {
              "uri": undefined,
            }
          }
          style={
            {
              "resizeMode": "cover",
            }
          }
        />
        <Text
          style={
            {
              "color": "#000",
              "fontSize": 22,
            }
          }
        >
          Charmander
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;

exports[`PokemonList renders list of pokemons correctly 1`] = `
<RCTScrollView
  data={
    [
      {
        "name": "Bulbasaur",
      },
      {
        "name": "Charmander",
      },
    ]
  }
  getItem={[Function]}
  getItemCount={[Function]}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={0.0001}
  stickyHeaderIndices={[]}
  testID="PokemonList"
  viewabilityConfigCallbackPairs={[]}
>
  <View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "borderColor": "#666",
            "borderTopWidth": 1,
            "flexDirection": "row",
            "justifyContent": "space-around",
            "opacity": 1,
            "padding": 15,
          }
        }
      >
        <Image
          source={
            {
              "uri": undefined,
            }
          }
          style={
            {
              "resizeMode": "cover",
            }
          }
        />
        <Text
          style={
            {
              "color": "#000",
              "fontSize": 22,
            }
          }
        >
          Bulbasaur
        </Text>
      </View>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "borderColor": "#666",
            "borderTopWidth": 1,
            "flexDirection": "row",
            "justifyContent": "space-around",
            "opacity": 1,
            "padding": 15,
          }
        }
      >
        <Image
          source={
            {
              "uri": undefined,
            }
          }
          style={
            {
              "resizeMode": "cover",
            }
          }
        />
        <Text
          style={
            {
              "color": "#000",
              "fontSize": 22,
            }
          }
        >
          Charmander
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
